src/index.js

```diff
@@ -438,70 +444,66 @@ class AvatarEditor extends React.Component {
     rotate = (rotate < 0) ? rotate + 360 : rotate
     rotate -= rotate % 90
 
-    const isPortrait = imageState.height > imageState.width
-
     if (this.state.mx && this.state.my) {
       const mx = this.state.mx - mousePositionX
       const my = this.state.my - mousePositionY
 
-      const xDiff = (rotate === 0 || rotate === 180 ? mx : my) / this.props.scale
-      const yDiff = (rotate === 0 || rotate === 180 ? my : mx) / this.props.scale
+      const width = this.state.image.width * this.props.scale
+      const height = this.state.image.height * this.props.scale
+
+      let {
+        x: lastX,
+        y: lastY
+      } = this.getCroppingRect()
+
+      lastX *= width
+      lastY *= height
+
+      const xDiff = (rotate === 0 || rotate === 180 ? mx : my)
+      const yDiff = (rotate === 0 || rotate === 180 ? my : mx)
 
       let y
       let x
 
       if (rotate === 0) {
-        y = lastY - yDiff
-        x = lastX - xDiff
+        y = lastY + yDiff
```

Comment:
Previously the x and y coordinates on the image referred to the x and y position of the image in the canvas - as they're now referring to the position of the cropping area within the image, the direction of movement gets reversed.